azurerm/resource_arm_backup_protected_file_share.go

```diff
@@ -0,0 +1,292 @@
+			SourceResourceID:  utils.String(storageAccountID),
+			FriendlyName:      utils.String(fileShareName),
+		},
+	}
+
+	resp, err := client.CreateOrUpdate(ctx, vaultName, resourceGroup, "Azure", containerName, protectedItemName, item)
+	if err != nil {
+		return fmt.Errorf("Error creating/updating Recovery Service Protected File Share %q (Resource Group %q): %+v", protectedItemName, resourceGroup, err)
+	}
+
+	locationURL, err := resp.Response.Location()
+	if err != nil || locationURL == nil {
+		return fmt.Errorf("Error creating/updating Azure File Share backup item %q (Vault %q): Location header missing or empty", containerName, vaultName)
+	}
+
+	opResourceID := azure.HandleAzureSdkForGoBug2824(locationURL.Path)
+
+	parsedLocation, err := azure.ParseAzureResourceID(opResourceID)
+	if err != nil {
+		return err
+	}
+	operationID := parsedLocation.Path["operationResults"]
+
+	if _, err := resourceArmBackupProtectedFileShareWaitForOperation(ctx, opClient, vaultName, resourceGroup, operationID, d); err != nil {
+		return err
+	}
+
+	resp, err = client.Get(ctx, vaultName, resourceGroup, "Azure", containerName, protectedItemName, "")
+
+	if err != nil {
+		return fmt.Errorf("Error creating/udpating Azure File Share backup item %q (Vault %q): %+v", protectedItemName, vaultName, err)
+	}
+
+	id := strings.Replace(*resp.ID, "Subscriptions", "subscriptions", 1) // This code is a workaround for this bug https://github.com/Azure/azure-sdk-for-go/issues/2824
+	d.SetId(id)
+
+	return resourceArmBackupProtectedFileShareRead(d, meta)
+}
+
+func resourceArmBackupProtectedFileShareRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).RecoveryServices.ProtectedItemsClient
+	ctx, cancel := timeouts.ForRead(meta.(*ArmClient).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	protectedItemName := id.Path["protectedItems"]
+	vaultName := id.Path["vaults"]
+	resourceGroup := id.ResourceGroup
+	containerName := id.Path["protectionContainers"]
+
+	log.Printf("[DEBUG] Reading Recovery Service Protected File Share %q (resource group %q)", protectedItemName, resourceGroup)
+
+	resp, err := client.Get(ctx, vaultName, resourceGroup, "Azure", containerName, protectedItemName, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on Recovery Service Protected File Share %q (Vault %q Resource Group %q): %+v", protectedItemName, vaultName, resourceGroup, err)
+	}
+
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("recovery_vault_name", vaultName)
+
+	if properties := resp.Properties; properties != nil {
+		if item, ok := properties.AsAzureFileshareProtectedItem(); ok {
+			sourceResourceID := strings.Replace(*item.SourceResourceID, "Microsoft.storage", "Microsoft.Storage", 1) // The SDK is returning inconsistent capitalization
```

Comment:
Is this still an issue?

Reply:
Yes, I believe so. It was causing warnings when I tried to run it without this. Here's an example response I get from the debug log...

```
2019-12-18T16: 02: 13.411-0600 [DEBUG] plugin.terraform-provider-azurerm: 
{
    "id": "/Subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/terraform-vault-test-rg/providers/Microsoft.RecoveryServices/vaults/test-vault-3/backupFabrics/Azure/protectionContainers/StorageContainer;storage;storage-rg;myteststorageaccount1234/protectedItems/AzureFileShare;test-share",
    "name": "AzureFileShare;test-share",
    "type": "Microsoft.RecoveryServices/vaults/backupFabrics/protectionContainers/protectedItems",
    "properties": {
        "friendlyName": "test-share",
        "protectionStatus": "Healthy",
        "protectionState": "Protected",
        "lastBackupStatus": "Completed",
        "lastBackupTime": "2019-12-18T16:30:40.0562496Z",
        "extendedInfo": {
            "oldestRecoveryPoint": "2019-12-18T16:31:09Z",
            "recoveryPointCount": 1,
            "policyState": "Consistent",
            "resourceState": "Active",
            "resourceStateSyncTime": "2019-12-17T22:09:26.7851504Z"
        },
        "protectedItemType": "AzureFileShareProtectedItem",
        "backupManagementType": "AzureStorage",
        "workloadType": "AzureFileShare",
        "containerName": "StorageContainer;storage;storage-rg;myteststorageaccount1234",
        "sourceResourceId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/storage-rg/providers/Microsoft.storage/storageAccounts/myteststorageaccount1234",
        "policyId": "/Subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/terraform-vault-test-rg/providers/Microsoft.RecoveryServices/vaults/test-vault-3/backupPolicies/tfex-recovery-vault-policy",
        "lastRecoveryPoint": "2019-12-18T16:31:09Z"
    }
}
```